### PR TITLE
Firewalld: Verify that the configuration has been read before writing

### PR DESCRIPTION
**Pull request by schubi2**
- https://bugzilla.suse.com/show_bug.cgi?id=1082827

Please review the following changes:
  * 1ee398d5 packaging
  * 29c57748 taking correct value for --set-log-denied

### CODE REVIEW

File: library/network/test/y2firewall/firewalld_test.rb [hunk 1]
```diff
@@ -262,7 +262,7 @@
         allow(zone).to receive(:modified?).and_return(modified_zone)
       end
       firewalld.zones = empty_zones
-      firewalld.log_denied_packets = true
+      firewalld.log_denied_packets = "on"
```
Comment (teclator) [inline library/network/test/y2firewall/firewalld_test.rb]: In that case it should be some of: all, unicast, broadcast, multicast

File: library/network/test/y2firewall/firewalld_test.rb [hunk 2]
```diff
@@ -357,7 +358,7 @@
 
     let(:api) do
       instance_double(Y2Firewall::Firewalld::Api,
-        log_denied_packets: true,
+        log_denied_packets: "on",
```
Comment (teclator) [inline library/network/test/y2firewall/firewalld_test.rb]: some as above

File: library/network/test/y2firewall/firewalld_test.rb [hunk 3]
```diff
@@ -378,7 +379,7 @@
       expect(config).to be_a(Hash)
       expect(config["enable_firewall"]).to eq(false)
       expect(config["start_firewall"]).to eq(true)
-      expect(config["log_denied_packets"]).to eq(true)
+      expect(config["log_denied_packets"]).to eq("on")
```
Comment (teclator) [inline library/network/test/y2firewall/firewalld_test.rb]: same as above

**Comment by teclator** [comment]
@schubi2 the bug in the description helps to identify the problem, could you attach it?. Also, the issue if I'm not wrong is because the value has not been initialized at all, isn't it?

**Comment by kobliha** [comment]
Just a question: what about the schema? Do we need to adapt it too? IMO we might need so. And the documentation as well, but that's a different PR.

**Comment by teclator** [comment]
@kobliha the problems arise when there is no firewall section.

**Comment by schubi2** [comment]
The docu is in well shape and the rnc file is checking for text only because the values are depending on the return value of the firewall-offline-cmd call.

**Comment by teclator** [comment]
@schubi2 I think that calling read during writing unless read? should be enough

File: package/yast2.changes [hunk 4]
```diff
@@ -1,3 +1,10 @@
+-------------------------------------------------------------------
+Tue Feb 27 15:19:25 CET 2018 - schubi@suse.de
+
+- Nis configuration fails while executing firewall-offline-cmd.
```
Comment (imobachgs) [inline package/yast2.changes]: np: NIS :)

**Pull request merged**